src/Analysis/Engine/Impl/Values/ProtocolInfo.cs

```diff
@@ -273,22 +268,10 @@ class ProtocolInfo : AnalysisValue, IHasRichDescription {
 
         public override int GetHashCode() => ObjectComparer.Instance.GetHashCode(_protocols);
 
-        internal override bool UnionEquals(AnalysisValue av, int strength) {
-            if (av is ProtocolInfo pi) {
-                if (strength > 0) {
-                    return Name == pi.Name;
-                }
-                return ObjectComparer.Instance.Equals(_protocols, pi._protocols);
-            }
-            return false;
-        }
+        internal override bool UnionEquals(AnalysisValue av, int strength)
```

Comment:
`av is ProtocolInfo pi && ObjectComparer.Instance.Equals(_protocols, pi._protocols)`